### ISSUE TITLE
Centred rotation rotating around the TopLeft

### ISSUE THREAD

**Issue by samuelwatsonofficial** [Weasy666/bevy_svg#34]
So my previous issue with the svg flicking from centred to topleft are gone but the svg acts as though it's origin is the top left rather than the centre. 
![image](https://github.com/Weasy666/bevy_svg/assets/84926729/4db7f293-c0c2-4d32-9942-e29d069a72fb)
![image](https://github.com/Weasy666/bevy_svg/assets/84926729/585b1a5f-c75b-42f4-926e-dbdbb75d7ab2)
![image](https://github.com/Weasy666/bevy_svg/assets/84926729/3c3db124-3d8c-4466-b965-44901bc2bfff)
if you want to check out the whole project where this is an issue look at https://github.com/samuelwatsonofficial/Scarlet/tree/main however the code concerned here is in character.rs:

#just for reference this code will not compile on it's own
```rust
let svg = asset_server.load("shapes/761.svg");
    commands
        .spawn(
            (
                RigidBody::Dynamic,
                GravityScale(5.),
                Collider::ball(10.),
                Velocity 
                {
                    linvel: Vec2::new(2.,2.),
                    angvel: 0.0,
                },
                Svg2dBundle
                {
                    svg,
                    origin: Origin::Center, #the rotation should be relative to the centre not the top left
                    transform: Transform
                    {
                        scale: Vec3::new(4.0, 4.0, 1.0),
                        translation: Vec3::new(0.0,3.0,0.0),
                        ..Default::default()
                    },
                    ..Default::default()
                },
                Character,
            ));
```

I'm fairly new to gamedev in general and very new to bevy so it's possible I've made an oversight but I've checked rapier and there should be no issues with centre of rotation on that end. I hope it's not too much effort to fix this, so far this plugin has been fantastic thanks for making it!

**Comment by Weasy666** [comment]
Hm... ok, that should not happen. The code snippet looks fine so far. I'll hopefully find some time on the weekend or next week. I'll take a look then. 🙂

**Comment by samuelwatsonofficial** [comment]
Fantastic, there's no rush because my project isn't even close to finished yet but that would be amazing of you. As for the flickering it just seemed to disappear as soon as I made a character controller, I'll see if I can test the old versions to find when it stopped.

**Comment by samuelwatsonofficial** [comment]
I just did a check on the old code I made and the flickering bug only occurs when there's nothing modifying the transform of the rapier rigidbody the SVG is attached to, in my case adding in a gravity scale was enough to completely fix this issue. It could be something to do with rapier2d knowing at compile time that nothing with modify the transform so there are optimisations that are causing unexpected behavior, but that's just a guess. Here are the two sample pieces of code to compare.

#No flickering 
pub fn spawn_character(mut commands: Commands, asset_server: Res<AssetServer>)
{
    println!("player exists now!");
    let svg = asset_server.load("shapes/761.svg");
    commands
        .spawn(
            (
                RigidBody::Dynamic,
                GravityScale(5.00),
                Collider::ball(10.),
                Svg2dBundle
                {
                    svg,
                    origin: Origin::Center,
                    transform: Transform
                    {
                        scale: Vec3::new(4.0, 4.0, 1.0),
                        translation: Vec3::new(0.0,3.0,0.0),
                        ..Default::default()
                    },
                    ..Default::default()
                }
            ));

}
#Flickering (notice the lack of a gravityscale)
pub fn spawn_character(mut commands: Commands, asset_server: Res<AssetServer>)
{
    println!("player exists now!");
    let svg = asset_server.load("shapes/761.svg");
    commands
        .spawn(
            (
                RigidBody::Dynamic,
                Collider::ball(10.),
                Svg2dBundle
                {
                    svg,
                    origin: Origin::Center,
                    transform: Transform
                    {
                        scale: Vec3::new(4.0, 4.0, 1.0),
                        translation: Vec3::new(0.0,3.0,0.0),
                        ..Default::default()
                    },
                    ..Default::default()
                }
            ));

}

**Comment by Weasy666** [comment]
Hm... i have no experience with rapier. Does rapier modify the transform of the entity it is attached to?

**Comment by Weasy666** [comment]
This should hopefully be fixed by #41, so i'll close this issue. If the problem still appears, please let me know.
